### PR TITLE
[SYCLomatic][codepin] Generate non-replacement processed file(s) in debug folder

### PR DESCRIPTION
**Pull request by zhiweij1**
Signed-off-by: Jiang, Zhiwei <zhiwei.jiang@intel.com>

### CODE REVIEW

**Review by wenhuiNi — APPROVED** [review]

File: clang/lib/DPCT/SaveNewFiles.cpp [hunk 1]
```diff
@@ -720,6 +723,9 @@ int saveNewFiles(clang::tooling::RefactoringTool &Tool,
       if (!rewriteDir(FilePath, InRoot, OutRoot)) {
         continue;
       }
+      if (!rewriteDir(DebugFilePath, InRoot, DebugCUDAFolder)) {
```
Comment (zhimingwang36) [inline clang/lib/DPCT/SaveNewFiles.cpp]: add isCodePinEnabled()?

File: clang/lib/DPCT/SaveNewFiles.cpp [hunk 2]
```diff
@@ -701,6 +702,8 @@ int saveNewFiles(clang::tooling::RefactoringTool &Tool,
   // "-out-root" directory.
   for (const auto &Entry : IncludeFileMap) {
     clang::tooling::UnifiedPath FilePath = Entry.first;
+    clang::tooling::UnifiedPath DebugFilePath = Entry.first;
```
Comment (zhimingwang36) [inline clang/lib/DPCT/SaveNewFiles.cpp:707]: comment the diff between the 3 FilePath

**Review by zhimingwang36 — CHANGES_REQUESTED** [review]
Please update the commit msg.

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**